### PR TITLE
Pickmove

### PR DESCRIPTION
**Pull request by gabriellaMT**

### CODE REVIEW

**Comment by digawp** [comment]
Have you tried compiling and running it (without the strategy calculation)? Does it work when you use `Game` instead of `State`?

File: PlayerSkeleton.java [hunk 1]
```diff
@@ -2,13 +2,28 @@
 public class PlayerSkeleton {
 
 	//implement this function to have a working system
-	public int pickMove(State s, int[][] legalMoves) {
+	public int pickMove(Game s, int[][] legalMoves) {
+		Strategy strategy = new Strategy();
```
Comment (vsharonlynn) [inline PlayerSkeleton.java]: Oops, my strategy takes in a double[] of weights.

File: PlayerSkeleton.java [hunk 2]
```diff
@@ -2,13 +2,28 @@
 public class PlayerSkeleton {
 
 	//implement this function to have a working system
-	public int pickMove(State s, int[][] legalMoves) {
+	public int pickMove(Game s, int[][] legalMoves) {
```
Comment (vsharonlynn) [inline PlayerSkeleton.java]: Could you add javadoc for this public method?

File: PlayerSkeleton.java [hunk 3]
```diff
@@ -2,13 +2,28 @@
...
+				move = i;
+			}
+		}
+		
+		return move;
 	}
 	
 	public static void main(String[] args) {
-		State s = new State();
+		Game s = new Game();
```
Comment (vsharonlynn) [inline PlayerSkeleton.java]: Have you considered of using subtyping, i.e. "State s = new Game();"? I wonder which one is of a better implementation?
Tag: @digawp @jials @nathanajah

File: Game.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+public class Game extends State implements Cloneable {
```
Comment (vsharonlynn) [inline Game.java]: Could add javadoc here? Just so we don't forget why we extend this class in the first place.

**Pull request merged**